src/utils/LookmlDiagrammer/join-utils.ts

```diff
@@ -41,14 +41,14 @@ export function getPkJoinPathObj(
   pkTableRef: DiagramField[],
   pkFieldIndex: number
 ) {
-  return {
+  return ({
     viewName: join.name,
     fieldIndex: pkFieldIndex,
-    selector: pkTableRef[pkFieldIndex].name.replace('.', '-'),
+    selector: pkTableRef[pkFieldIndex]?.name?.replace('.', '-'),
     type: 'core',
     joinName: join.name,
     joinObj: join,
-  }
+  } as unknown) as DiagramField
```

Comment:
Guessing this came from internal testing?

Reply:
the PR is in response to a customer ticket, if that's what you mean?

Reply:
Ah i meant this specific line's type casting.